futures-executor/src/thread_pool.rs

```diff
@@ -315,8 +316,8 @@ impl Task {
                 };
                 match res {
                     Ok(Async::Pending) => {}
-                    Ok(Async::Ready(())) |
-                    Err(()) => return wake_handle.mutex.complete(),
+                    Ok(Async::Ready(())) => return wake_handle.mutex.complete(),
+                    Err(_) => unreachable!(), // FIXME: This case shouldn't be needed at all!
```

Comment:
I was under the impression that rust no longer required patterns containing uninhabited types to be matched, but it seems like that isn't the case yet?

Reply:
Have you tried `Err(never) => match never {}` here?

Reply:
@Ralith That change got rolled back and is being separately feature gated from the `!` type due to backcompat concerns. IMO it'll probably still go through at some point, but it isn't available now. For the time being, I'd use `match never {}` as @srijs suggested.